File: JCPhotoBrowser/JCPhotoItem.m
//
//  JCPhotoItem.m
//  JCPhotoBrowser
//
//  Created by Jake Cai on 12/25/16.
//  Copyright © 2016 Jake Cai. All rights reserved.
//

#import "JCPhotoItem.h"

@interface JCPhotoItem ()

@property (nonatomic, strong, readwrite) UIView *sourceView;

@property (nonatomic, strong, readwrite) UIImage *thumbImage;

@property (nonatomic, strong, readwrite) UIImage *image;

@property (nonatomic, strong, readwrite) NSURL *imageUrl;

@end

@implementation JCPhotoItem

- (instancetype)initWithSourceView:(UIView *)view
                        thumbImage:(UIImage *)image
                          imageUrl:(NSURL *)url {
    self = [super init];
    if (self) {
        _sourceView = view;
        _thumbImage = image;
        _imageUrl = url;
    }
    return self;
}

- (instancetype)initWithSourceView:(UIImageView *)view
                          imageUrl:(NSURL *)url {
    return [self initWithSourceView:view
                         thumbImage:view.image
                           imageUrl:url];
}

- (instancetype)initWithSourceView:(UIImageView *)view
                             image:(UIImage *)image {
    self = [super init];
    if (self) {
        _sourceView = view;
        _thumbImage = image;
        _imageUrl = nil;
        _image = image;
    }
    return self;
}

+ (instancetype)itemWithSourceView:(UIView *)view
                         thumbImage:(UIImage *)image
                           imageUrl:(NSURL *)url {
    return [[JCPhotoItem alloc] initWithSourceView:view
                                    thumbImage:image
                                      imageUrl:url];
}

+ (instancetype)itemWithSourceView:(UIImageView *)view
                           imageUrl:(NSURL *)url {
    return [[JCPhotoItem alloc] initWithSourceView:view
                                      imageUrl:url];
}

+ (instancetype)itemWithSourceView:(UIImageView *)view
                              image:(UIImage *)image {
    return [[JCPhotoItem alloc] initWithSourceView:view
                                         image:image];
}

@end

File: JCPhotoBrowser-swift/JCPhotoItem.swift
//
//  JCPhotoItem.swift
//  JCPhotoBrowser-Swift
//
//  Created by Jake on 31/05/2017.
//  Copyright © 2017 Jake. All rights reserved.
//

import UIKit


class JCPhotoItem: NSObject {
    var sourceView:UIView!
    var thumbImage:UIImage!
    var image:UIImage!
    var imageUrl:URL!
    var finished:Bool! = false
    

    @objc public init(_ source:UIView?,_ thumb:UIImage?,_ url:URL?) {
        super.init()
        self.sourceView = source
        self.thumbImage = thumb
        self.imageUrl = url
    }
    
    @objc public convenience init(_ source:UIImageView?,_ url:URL?) {
        self.init(source,source?.image,url)
    }
    
    @objc public init(_ source:UIImageView?,_image:UIImage?) {
        super.init()
        self.sourceView = source
        self.thumbImage =  _image
        self.imageUrl = nil
        self.image = _image
    }
    
}
